lib/controller.dart

```diff
@@ -149,6 +143,91 @@ Map<String, Set> computeUniqFieldCategoryValues(
   return uniqueFieldCategories;
 }
 
+bool _interactionMatchesFilters(
+    Map<String, dynamic> interaction, Map<String, String> filters) {
+  for (var entry in filters.entries) {
+    var interactionValue = interaction[entry.key];
+    var interactionMatch = interactionValue is List
+        ? interactionValue.contains(entry.value)
+        : interactionValue == entry.value;
+
+    if (!interactionMatch) {
+      return false;
+    }
+  }
+
+  return true;
+}
+
+bool _interactionMatchesOperation(
+    Map<String, dynamic> interaction, model.Field chartCol) {
+  switch (chartCol.field.operator) {
+    case model.FieldOperator.equals:
+      if (interaction[chartCol.field.key] == chartCol.field.value) {
+        return true;
+      }
+      break;
+    case model.FieldOperator.contains:
+      if ((interaction[chartCol.field.key] as List)
+          .contains(chartCol.field.value)) {
+        return true;
+      }
+      break;
+    case model.FieldOperator.not_contains:
+      if (!(interaction[chartCol.field.key] as List)
+          .contains(chartCol.field.value)) {
+        return true;
+      }
+      break;
+    default:
+      logger.error('No such operator, misleading results');
+      view.showAlert(
+          'Operator ${chartCol.field.operator} is not handled in your config!');
+  }
```

Comment:
I think the info that results may be misleading is a message for the user, not the log. It would also be useful to log the operator, not just show it to the user. How about something like: 
```suggestion
      logger.error('No such operator: ${chartCol.field.operator}');
      view.showAlert('Warning: Field operator ${chartCol.field.operator} listed in your config is not supported. Results may be misleading');
  }
```

We may also want to analyse the user config when the user sets it, so that we show the error earlier in the process (so that the user can change the operator), in which case this error shouldn't happen (but we should still keep it in in case it does!)

Reply:
Ah sure, will make note for the user config when they try to set it.